src/System.Net.Http/src/System/Net/Http/Managed/BasicAuthenticationHelper.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+
+namespace System.Net.Http
+{
+    internal static class BasicAuthenticationHelper
+    {
+        public static string GetBasicTokenForCredential(NetworkCredential credential)
+        {
+            if (credential.UserName.IndexOf(':') != -1)
+            {
+                // TODO: What's the right way to handle this?
+                //              throw new NotImplementedException($"Basic auth: can't handle ':' in username \"{credential.UserName}\"");
+            }
+
+            string userPass = credential.UserName + ":" + credential.Password;
+            if (!string.IsNullOrEmpty(credential.Domain))
+            {
+                if (credential.UserName.IndexOf(':') != -1)
+                {
+                    // TODO: What's the right way to handle this?
+                    //                  throw new NotImplementedException($"Basic auth: can't handle ':' in domain \"{credential.Domain}\"");
```

Comment:
(Better than looking for a : to fail is to assert that only legal values are used)